docs/platform-branches.md

```diff
@@ -0,0 +1,98 @@
+<!--
+     Copyright 2024, Proofcraft Pty Ltd
+     SPDX-License-Identifier: CC-BY-SA-4.0
+-->
+
+# Automated Platform Branches
+
+The following branches for additional platforms are currently automatically
+rebased and tested on each push to `master`:
+
+- [imx8mm-fpu-ver](https://github.com/seL4/l4v/tree/imx8-fpu-ver)
+- [exynos5-ver](https://github.com/seL4/l4v/tree/exynos5-ver)
+
+## How this works
+
+The `L4V_PLAT` configuration setting provides an additional dimension to the
+configuration space for verified platforms in addition to `L4V_ARCH` and
+`L4V_FEATURES`. To test an `L4V_ARCH`, `L4V_FEATURES`, and `L4V_PLAT`
+combination, the test setup will attempt to load a corresponding kernel config
+from the seL4 repository under
+
+```sh
+configs/${L4V_ARCH}_${L4V_PLAT}_${L4V_FEATURES}_verified.cmake
+```
+
+e.g. `ARM_HYP_exynos5_verified.cmake`. Tests on the master branch will currently
+keep `L4V_PLAT` empty, and tests on the platform branches will provide the
+appropriate platform string.
+
+To keep platform branches up to date, there is [a GitHub action][rebase-action]
+that automatically attempts to rebase all platform branches over each new push
+to the master branch. If that rebase succeeds, this action will push the rebased
+platform branch to a candidate branch `<branch>-rebased`. The platform branch,
+and therefore the `-rebased` branch have a corresponding action that in turn is
+triggered by this push and runs a full proof test for this branch. If the test
+succeeds, the branch action replaces the original platform branch with the
+candidate `<branch>-rebased` branch by force push. If either step fails, CI will
+signal test failure and require manual intervention.
+
+[rebase-action]: https://github.com/seL4/l4v/blob/d869bb5ec81/.github/workflows/proof-deploy.yml#L89
+
+## How to fix things if they go wrong
+
+### Invariants
+
+- you should never have to modify the original platform branches
+
+- the original platform branches are updated by CI only
+
+- therefore, the original platform branches always are in a state that has
+  worked before with a specific seL4 version (the last one before CI signalled
+  failure)
+
+- the `-rebased` branches are ephemeral and can be force-pushed to
+
+### Rebase fails
+
+If the rebase action fails, for instance because there are conflicts with the
+master branch during `git rebase`, the manual steps to fixing the problem are
+for example:
+
+```sh
+# get up-to-date branches:
+git fetch
+# start from the old platform branch:
+git checkout verification/exynos5-ver
+# prepare a -rebased branch for it
+# The name is important, you may have delete an old one if it already exists:
+git checkout -b exynos5-ver-rebased
+# rebase over current master:
+git rebase verification/master
+# manually fix rebase conflicts
+...
+# usually you want to check if evervthing works --
+# be sure to select correct L4V_ARCH and L4V_PLAT
+L4V_ARCH=ARM_HYP L4V_PLAT=exynos5 ./run_tests
```

Comment:
this is a bit off-topic for this PR, but the C kernel doesn't get regenerated if you change environment variables, right? So if someone had C code sitting around from master, and then ran the above command, would that mean everything would work up until CRefine broke? If so, it it worth to remind people at this point in the doc?

Reply:
I actually need to look into this a bit more to understand when exactly it will miss rebuilds. It does change the cmake file it is looking for, so something visible to `make` does change. Maybe what I should be doing is to also change the build directory name (i.e. encode L4V_PLAT in it, like we do L4V_ARM) -- I think then we might get the correct rebuilds. That'd be overall nicer. Will investigate.

Reply:
Ok, the wrinkle is that we do `install_C_file "../c/build/$L4V_ARCH/kernel_all.c_pp"` in `Kernel_C.thy`, and I don't think I can do any conditional magic in Isabelle there. We could set something like

```
KERNEL_BUILD_ROOT := build/${L4V_ARCH}$(if ${L4V_PLAT},_${L4V_PLAT},)
```

in the Makefile that calls the session `CKernel`, since that already needs to make sure it builds the C code first, and then use

```
install_C_file "../c/$KERNEL_BUILD_ROOT/kernel_all.c_pp
```

in `Kernel_C`. It will make everything one level more obscure, but it'd work.

Reply:
We could avoid conditionals and Makefile magic  if we accept `c/build/ARM_/` as the name of the build directory if there is no L4V_PLAT set and do

```
install_C_file "../c/build/$L4V_ARCH_$L4V_PLAT/kernel_all.c_pp"
```

in `Kernel_C`. Less pretty, but a lot less obscure. I might give that a shot for a PR.

Reply:
This was a deeper rabbit hole than I thought. I'll adjust the description here for now and we can discuss more in #762.